src/CloudNative.CloudEvents/CloudEventsSpecVersion.cs

```diff
@@ -1,15 +1,146 @@
-﻿// Copyright (c) Cloud Native Foundation. 
+﻿// Copyright 2021 Cloud Native Foundation. 
 // Licensed under the Apache 2.0 license.
 // See LICENSE file in the project root for full license information.
 
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Mime;
+using static CloudNative.CloudEvents.CloudEventAttribute;
+
 namespace CloudNative.CloudEvents
 {
-    public enum CloudEventsSpecVersion
+    // TODO: Rename to CloudEventSpecVersion? Using the plural form feels a little odd.
+
+    /// <summary>
+    /// Represents a version of the CloudEvents specification, including
+    /// the context attribute values known to that version.
+    /// </summary>
+    public sealed class CloudEventsSpecVersion
     {
-        V0_1,
-        V0_2,
-        V0_3,
-        V1_0,
-        Default = V1_0
+        /// <summary>
+        /// The attribute used to indicate the version of the CloudEvents specification being used.
+        /// </summary>
+        public static CloudEventAttribute SpecVersionAttribute { get; } = CreateRequired("specversion", CloudEventAttributeType.String, NonEmptyString);
+
+        // Populated by the constructor.
+        private static readonly List<CloudEventsSpecVersion> allVersions = new List<CloudEventsSpecVersion>();
+
+        public static CloudEventsSpecVersion Default => V1_0;
```

Comment:
Would Default only update for new major spec versions (along with a new major SDK version)?

Reply:
Probably? :)

It's really hard to know how all the versioning stuff will pan out when we basically don't have any data points. (I don't think we can infer much from what happened pre-1.0, as obviously more attention will be paid to compatibility post 1.0.)